File: sbe/binance/spot_3_2/generated-python/spot_sbe/UserDataStreamSubscribeResponse.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple, Set
from io import BytesIO

class UserDataStreamSubscribeResponse:
    """SBE message: UserDataStreamSubscribeResponse."""

    TEMPLATE_ID = 503
    SCHEMA_ID = 3
    SCHEMA_VERSION = 2
    BLOCK_LENGTH = 2

    def __init__(self):
        self.subscription_id = None

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.subscription_id is not None:
            buffer.write(struct.pack('<H', self.subscription_id))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        pos = 0

        self.subscription_id = struct.unpack_from('<H', data, pos)[0]
        pos += 2

        # Skip to end of block for forward compatibility
        pos = 2

File: sbe/binance/spot_3_2/generated-php/spot_sbe/UserDataStreamSubscribeResponse.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class UserDataStreamSubscribeResponse
{
    public const TEMPLATE_ID = 503;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 2;
    public const BLOCK_LENGTH = 2;

    public int|float|array|null $subscriptionId = null;

    public function encode(): string
    {
        $buffer = '';

        if ($this->subscriptionId !== null) {
            $buffer .= pack('v', $this->subscriptionId);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->subscriptionId = unpack('v', substr($data, $offset, 2))[1];
        $offset += 2;

        // Skip to end of block for forward compatibility
        $offset = 2;

    }
}
